lib/server.js

```diff
@@ -20,8 +20,11 @@ exports.live = function(config, root) {
 };
 
 function parse(config) {
-	// eslint-disable-next-line no-unused-vars
-	let [_, host, port] = CONFIG_FORMAT.exec(config);
-	host = host ? host.slice(0, -1) : "0.0.0.0";
-	return [host, port];
+	let port = PORT_PATTERN.exec(config);
+    port = port && port[1];
+	if(port) { // `config` contains both host and port
+		host = config.substr(0, config.length - port.length - 1);
+		return [host, port];
```

Comment:
Would donny or live-server balk at `parseInt(port, 10)`?

Reply:
Could we not simplify this entire thing like this:

```js
function parse(config) {
  if(config.includes(":")) {
    return config.split(":");
  }
  // the user only provided a port
  return ["0.0.0.0", config];
}
```

parseInt:
donny doesn't care. I guess live-server doesn't either. Why do you ask?

Reply:
I'd briefly considered simplifying it like that, but didn't wanna encode assumptions about the host IP/name - think IPv6. IOW "ends in colonumbers" seemed more precise/accurate/safe.

`parseInt`: I like to use integers for port numbers, also because it acts as a bit of a sanity check. Doesn't really matter though.

Reply:
Ok, then let me suggest:

```js
function parse(config) {
  let seperator = config.lastIndexOf(":");
  if(seperator) {
    let host = config.substr(0, seperator);
    let port = config.substr(seperator + 1);
    return [host, port];
  }
  return ["0.0.0.0", config];
}
```

Reply:
TIL `lastIndexOf` 😳 

Yeah, I like it: 45a9973cf81114c2ed7c899b427beb06ec1a2a6c

I've only done some cursory testing though:
```javascript
> parse("999")
[ '0.0.0.0', '999' ]
> parse(":999")
[ '', '999' ]
> parse("local:host::999")
[ 'local:host:', '999' ]
```